### DISCUSSION TITLE
beside ubuntu, RedHat 8/9 or Centos8.

### DISCUSSION THREAD

**Discussion by louis-ovt** [The-OpenROAD-Project/OpenROAD#5732] [Q&A]
Hi we install OpenRoad in ubuntu but failed to install in RedHat 8/9 or Centos8. is it true that ubuntu is the only one where we can install OpenRoad to?

**Comment by maliberty** [comment]
We build and test on 

Ubuntu 20.04/22.04
Debian 10/11

We plan to add RHEL in the future.  What errors are you seeing?

**Comment by vikashpatel24** [comment]
Hello sir, I am getting error while installing OpenROAD on RHEL 8.10 and in CENTOS 9. For both OS, I am getting stuck after 67% of building. I am getting the following errors:

[ 38%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/opt/ret/retFlow.c.o
[ 38%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/opt/ret/retIncrem.c.o
In file included from /usr/local/include/spdlog/fmt/bundled/format.h:41,
                 from /usr/local/include/spdlog/fmt/bundled/core.h:5,
                 from /usr/local/include/spdlog/fmt/fmt.h:24,
                 from /usr/local/include/spdlog/common.h:50,
                 from /usr/local/include/spdlog/details/os.h:7,
                 from /edacloud/tools/OpenROAD-flow-scripts/tools/OpenROAD/src/utl/include/utl/Logger.h:51,
                 from /edacloud/tools/OpenROAD-flow-scripts/tools/OpenROAD/src/odb/include/odb/odb.h:43,
                 from /edacloud/tools/OpenROAD-flow-scripts/tools/OpenROAD/src/odb/include/odb/ZException.h:38,
                 from /edacloud/tools/OpenROAD-flow-scripts/tools/OpenROAD/src/odb/src/db/dbAttrTable.h:35,
                 from /edacloud/tools/OpenROAD-flow-scripts/tools/OpenROAD/src/odb/src/db/dbCore.h:47,
                 from /edacloud/tools/OpenROAD-flow-scripts/tools/OpenROAD/src/odb/src/db/dbMarker.h:36,
                 from /edacloud/tools/OpenROAD-flow-scripts/tools/OpenROAD/src/odb/src/db/dbMarker.cpp:34:
/usr/local/include/spdlog/fmt/bundled/base.h: In instantiation of ‘constexpr fmt::v11::detail::value<Context> fmt::v11::detail::make_arg(T&) [with bool PACKED = true; Context = fmt::v11::context; T = const boost::optional<std::__cxx11::basic_string<char> >; typename std::enable_if<PACKED, int>::type <anonymous> = 0]’:
/usr/local/include/spdlog/fmt/bundled/base.h:2018:74:   required from ‘constexpr fmt::v11::detail::format_arg_store<Context, NUM_ARGS, 0, DESC> fmt::v11::make_format_args(T& ...) [with Context = fmt::v11::context; T = {const char* const, const char* const, int, const boost::optional<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >}; long unsigned int NUM_ARGS = 4; long unsigned int NUM_NAMED_ARGS = 0; long long unsigned int DESC = 61900; typename std::enable_if<(NUM_NAMED_ARGS == 0), int>::type <anonymous> = 0]’
/usr/local/include/spdlog/logger.h:328:75:   required from ‘void spdlog::logger::log_(spdlog::source_loc, spdlog::level::level_enum, spdlog::string_view_t, Args&& ...) [with Args = {const char* const&, const char* const&, int&, const boost::optional<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >&}; spdlog::string_view_t = fmt::v11::basic_string_view<char>]’
/usr/local/include/spdlog/logger.h:80:13:   required from ‘void spdlog::logger::log(spdlog::source_loc, spdlog::level::level_enum, fmt::v11::format_string<T ...>, Args&& ...) [with Args = {const char* const&, const char* const&, int&, const boost::optional<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >&}; fmt::v11::format_string<T ...> = fmt::v11::basic_format_string<char, const char* const&, const char* const&, int&, const boost::optional<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >&>]’
/usr/local/include/spdlog/logger.h:85:12:   required from ‘void spdlog::logger::log(spdlog::level::level_enum, fmt::v11::format_string<T ...>, Args&& ...) [with Args = {const char* const&, const char* const&, int&, const boost::optional<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >&}; fmt::v11::format_string<T ...> = fmt::v11::basic_format_string<char, const char* const&, const char* const&, int&, const boost::optional<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >&>]’
/edacloud/tools/OpenROAD-flow-scripts/tools/OpenROAD/src/utl/include/utl/Logger.h:272:19:   required from ‘void utl::Logger::log(utl::ToolId, spdlog::level::level_enum, int, const string&, const Args& ...) [with Args = {boost::optional<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >}; std::string = std::__cxx11::basic_string<char>]’
/edacloud/tools/OpenROAD-flow-scripts/tools/OpenROAD/src/utl/include/utl/Logger.h:166:8:   required from ‘void utl::Logger::warn(utl::ToolId, int, const string&, const Args& ...) [with Args = {boost::optional<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >}; std::string = std::__cxx11::basic_string<char>]’
/edacloud/tools/OpenROAD-flow-scripts/tools/OpenROAD/src/odb/src/db/dbMarker.cpp:509:26:   required from here
/usr/local/include/spdlog/fmt/bundled/base.h:1641:63: error: ‘fmt::v11::detail::type_is_unformattable_for<const boost::optional<std::__cxx11::basic_string<char> >, char> _’ has incomplete type
 1641 |     type_is_unformattable_for<T, typename Context::char_type> _;
      |                                                               ^
/usr/local/include/spdlog/fmt/bundled/base.h:1644:7: error: static assertion failed: Cannot format an argument. To make type T formattable provide a formatter<T> specialization: https://fmt.dev/latest/api.html#udt
 1644 |       formattable,
      |       ^~~~~~~~~~~
/usr/local/include/spdlog/fmt/bundled/base.h:1644:7: note: ‘formattable’ evaluates to false
[ 38%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/opt/ret/retInit.c.o
gmake[2]: *** [src/odb/src/db/CMakeFiles/db.dir/build.make:1532: src/odb/src/db/CMakeFiles/db.dir/dbMarker.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
[ 38%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/opt/ret/retLvalue.c.o      

and   


[ 65%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/bbr/bbrCex.c.o
[ 65%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/bbr/bbrImage.c.o
[ 65%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/bbr/bbrNtbdd.c.o
[ 65%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/bbr/bbrReach.c.o
[ 65%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/llb/llb1Cluster.c.o
[ 67%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/llb/llb1Constr.c.o
[ 67%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/llb/llb1Core.c.o
[ 67%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/llb/llb1Group.c.o
[ 67%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/llb/llb1Hint.c.o
[ 67%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/llb/llb1Man.c.o
[ 67%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/llb/llb1Matrix.c.o
[ 67%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/llb/llb1Pivot.c.o
[ 67%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/llb/llb1Reach.c.o
[ 67%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/llb/llb1Sched.c.o
[ 67%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/llb/llb2Bad.c.o
[ 67%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/llb/llb2Core.c.o
[ 67%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/llb/llb2Driver.c.o
[ 67%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/llb/llb2Dump.c.o
[ 67%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/llb/llb2Flow.c.o
[ 67%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/llb/llb2Image.c.o
[ 67%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/llb/llb3Image.c.o
[ 67%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/llb/llb3Nonlin.c.o
[ 67%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/llb/llb4Cex.c.o
[ 67%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/llb/llb4Image.c.o
[ 67%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/llb/llb4Nonlin.c.o
[ 67%] Building CXX object third-party/abc/CMakeFiles/libabc.dir/src/bdd/llb/llb4Sweep.c.o
[ 67%] Linking CXX static library libabc.a
[ 67%] Built target libabc
gmake: *** [Makefile:146: all] Error 2

real    5m32.064s
user    19m15.747s
sys     2m16.804s

**Reply by vikashpatel24** [reply]
Sir, Please help me in resolving these errors

**Reply by vikashpatel24** [reply]
In the following link it's written that OpenROAD supports RHEL. Here's the supported link:  https://openroad.readthedocs.io/en/latest/index.html#system-requirements

**Reply by vikashpatel24** [reply]
I tried with the following also: https://openroad-flow-scripts.readthedocs.io/en/latest/user/BuildLocally.html , but it also failed after 67% of building with same errors. Yosys was installed properly with this method and standalone installation of yosys is also working properly on both RHEL AND CENTOS.

**Comment by maliberty** [comment]
Probably you are on a newer version of spdlog.

**Reply by vikashpatel24** [reply]
So sir, which version of spdlog is suitable for rhel 8.10 or CentOS 9?

**Reply by maliberty** [reply]
1.8.1 
https://github.com/The-OpenROAD-Project/OpenROAD/blob/35d3dd3e024f43af6198eea27a711585a74aa051/etc/DependencyInstaller.sh#L75

**Comment by vikashpatel24** [comment]
Ok sir, I will try with this version and if I will get any problem I will
write in this thread. Thank you sir
